operator/v1/types_olm.go

```diff
@@ -0,0 +1,57 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// OLM provides information to configure an operator to manage the OLM controllers
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=olms,scope=Cluster
+// +kubebuilder:subresource:status
+// +openshift:api-approved.openshift.io=https://github.com/openshift/api/pull/1504
+// +openshift:file-pattern=cvoRunLevel=0000_10,operatorName=operator-lifecycle-manager,operatorOrdering=01
+// +kubebuilder:validation:XValidation:rule="self.metadata.name == 'cluster'",message="olm is a singleton, .metadata.name must be 'cluster'"
```

Comment:
Has this been tested and we are happy that it works?

Reply:
yup yup - we have a test for it https://github.com/openshift/api/pull/2061/files#diff-34775f0b3bbcc2a64f685c6fc398e13b3fae20851398b2b5a455726630673a01R21

I've also gone ahead and manually tested on a clusterbot. Looks good ^^